src/services/ui/src/pages/form/modals.tsx

```diff
@@ -0,0 +1,86 @@
+import { ConfirmationModal } from "@/components";
```

Comment:
Have you checked out `createContextProvider`?
```js
import { createContextProvider } from "@/utils";
```

Here's how I would implement with it:

```js
const Success = () => {
  const navigate = useNavigate();
  const { setSuccessModalOpen, successModalOpen } = useModalContext();
  return (
    <ConfirmationModal
      open={successModalOpen}
      onAccept={() => {
        setSuccessModalOpen(false);
        navigate(ROUTES.DASHBOARD);
      }}
      onCancel={() => setSuccessModalOpen(false)}
      title="Submission Successful"
      body={
        <p>
          Please be aware that it may take up to a minute for your submission to
          show in the Dashboard.
        </p>
      }
      cancelButtonVisible={false}
      acceptButtonText="Go to Dashboard"
    />
  );
};

const Cancel = () => {
  const navigate = useNavigate();
  const { setCancelModalOpen, cancelModalOpen } = useModalContext();
  return (
    <ConfirmationModal
      open={cancelModalOpen}
      onAccept={() => {
        setCancelModalOpen(false);
        navigate(ROUTES.DASHBOARD);
      }}
      onCancel={() => setCancelModalOpen(false)}
      cancelButtonText="Return to Form"
      acceptButtonText="Yes"
      title="Are you sure you want to cancel?"
      body={
        <p>If you leave this page you will lose your progress on this form</p>
      }
    />
  );
};

const useModal = () => {
  const [cancelModalOpen, setCancelModalOpen] = useState(false);
  const [successModalOpen, setSuccessModalOpen] = useState(false);
  return {
    cancelModalOpen,
    setCancelModalOpen,
    successModalOpen,
    setSuccessModalOpen,
  };
};

export const [ModalContextProvider, useModalContext] = createContextProvider<
  ReturnType<typeof useModal>
>({
  name: "OsSearch Context",
  errorMessage: "forgot to wrap with OsProvider",
});

export const ModalProvider = ({ children }: PropsWithChildren) => {
  return (
    <ModalContextProvider value={useModal()}>
      {children}
      <Success />
      <Cancel />
    </ModalContextProvider>
  );
};
```